components/google-cloud/google_cloud_components/aiplatform/remote_runner.py

```diff
@@ -15,140 +15,186 @@
+    """Splits args into constructor and method args.
 
-  gcs_uri = gcs_uri[len('gs://'):]
-  gcs_bucket, gcs_blob = gcs_uri.split('/', 1)
+    Args:
+        kwargs: kwargs with parameter names preprended with init or method
+    Returns:
+        constructor kwargs, method kwargs
+    """
+    init_args = {}
+    method_args = {}
 
-  with open('resource_name.txt', 'w') as f:
-    f.write(text)
+    for key, arg in kwargs.items():
+        if key.startswith(INIT_KEY):
+            init_args[key.split(".")[-1]] = arg
+        elif key.startswith(METHOD_KEY):
+            method_args[key.split(".")[-1]] = arg
 
-  client = storage.Client(project=project)
-  bucket = client.get_bucket(gcs_bucket)
-  blob = bucket.blob(os.path.join(gcs_blob, 'resource_name.txt'))
-  blob.upload_from_filename('resource_name.txt')
+    return init_args, method_args
 
 
-def read_from_gcs(project, gcs_uri):
-  gcs_uri = gcs_uri[len('gs://'):]
-  gcs_bucket, gcs_blob = gcs_uri.split('/', 1)
-  client = storage.Client(project=project)
-  bucket = client.get_bucket(gcs_bucket)
-  blob = bucket.get_blob(os.path.join(gcs_blob, 'resource_name.txt'))
-  resource_name = blob.download_as_string().decode('utf-8')
-  return resource_name
+def write_to_gcs(project, gcs_uri, text):
+    """helper method to write to gcs
+    TODO: remove and use native support from Pipeline
+    """
+    gcs_uri = gcs_uri[5:]
+    gcs_bucket, gcs_blob = gcs_uri.split('/', 1)
 
+    with open('resource_name.txt', 'w') as f:
+        f.write(text)
 
-def resolve_project(serialized_args):
-  return serialized_args[INIT_KEY].get(
-      'project', serialized_args[METHOD_KEY].get('project'))
+    client = storage.Client(project=project)
+    bucket = client.get_bucket(gcs_bucket)
+    blob = bucket.blob(os.path.join(gcs_blob, 'resource_name.txt'))
+    blob.upload_from_filename('resource_name.txt')
 
 
-def resolve_input_args(value, inpu_type, project):
-  """If this is an input from Pipelines, read it directly from gcs."""
-  if issubclass(inpu_type, aiplatform.base.AiPlatformResourceNoun):
-    if value.startswith('gs://'):  # not a resource noun:
-      value = read_from_gcs(project, value)
-  return value
+def read_from_gcs(project, gcs_uri):
+    """helper method to read from gcs
+    TODO: remove and use native support from Pipelines
+    """
+    gcs_uri = gcs_uri[5:]
+    gcs_bucket, gcs_blob = gcs_uri.split('/', 1)
+    client = storage.Client(project=project)
+    bucket = client.get_bucket(gcs_bucket)
+    blob = bucket.get_blob(os.path.join(gcs_blob, 'resource_name.txt'))
+    resource_name = blob.download_as_string().decode('utf-8')
+    return resource_name
+
+
+def resolve_project(serialized_args: Dict[str, Dict[str, Any]]) -> str:
+    """Gets the project from either constructor or method.
+    """
+    return serialized_args['init'].get(
+        'project', serialized_args['method'].get('project')
+    )
+
+
+def resolve_input_args(value, _type, project):
```

Comment:
Nit: I'm not sure `_type` is conformant to the style guide.

Reply:
Done. Thanks!